code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -150,13 +144,12 @@
 	if(!on_fire)
 		return TRUE
 
-	if(isanimal(owner))
-		var/mob/living/simple_animal/animal_owner = owner
-		adjust_stacks(animal_owner.fire_stack_removal_speed * delta_time)
+	if(isbasicmob(owner))
+		adjust_stacks(BASIC_MOB_FIRE_DECAY_RATE * delta_time)
```

Comment:
On this I wondered if it might be better to just make `fire_decay_rate` a `var` on `mob/living` and just read it instead of doing this type checking here. Not sure if that's an improvement or if it's bloating the type, would appreciate a maintainer opinion.

Reply:
if all mobs could catch on fire that'd be useful but aren't we lacking fire implementation on some mobs still?

Reply:
Yes, _most_ mobs in fact. It's basically just humans, cyborgs, xenos, and spiders which is why I was not sure it is worthwhile.
I wouldn't mind extending it to more mobs which make sense in the future but obviously can't do that this month, but until then I just didn't like this `if... else if... else` very much either even though it does everything that we need it to do.

I wasn't sure if the time to adjust it is when we're checking against 3 things, or against 4.

Reply:
if it's already implemented between `silicon`, `basic`, and `carbon` (xenos and humans) i think it should be done now tbqh as that's our three major domains of life in-game... the only issue would probably be `brain`, since that's some sort of abstract bullshit that counts as "living", but we can always override it on that subtype level.

Reply:
at some point whatever variable cost that we're saving _probably_ isn't worth all that much seeing that:

90% of any instantiated mob in a game has the propensity to catch on fire or is just abstract anyways
it's a potential dampener to a much more simplistic/elegant solution

the first one sounds alright to me, and if turning it into a variable could result in such a solution i don't see why it shouldn't be done. it's likely not the end of the world if we just keep tacking stuff in an if/else chain but at some point it stops making sense to not pursue the cleaner option